MatrixSDK/Crypto/Verification/MXKeyVerificationManager.m

```diff
@@ -944,7 +947,25 @@ - (void)handleSASKeyVerificationStart:(MXSASKeyVerificationStart*)keyVerificatio
     [self loadDeviceWithDeviceId:keyVerificationStart.fromDevice andUserId:event.sender success:^(MXDeviceInfo *otherDevice) {
         
         MXKeyVerificationTransaction *existingTransaction = [self transactionWithUser:event.sender andDevice:keyVerificationStart.fromDevice];
-        if (existingTransaction)
+        
+        MXQRCodeTransaction *existingQRCodeTransaction;
+        
+        BOOL isExistingTransactionTemporary = NO;
+        
+        if ([existingTransaction isKindOfClass:MXQRCodeTransaction.class])
+        {
+            existingQRCodeTransaction = (MXQRCodeTransaction*)existingTransaction;
+            
+            if (existingQRCodeTransaction.state == MXQRCodeTransactionStateUnknown)
+            {
+                isExistingTransactionTemporary = YES;
+                
+                // Remove fake QR code transaction
+                [self removeQRCodeTransactionWithTransactionId:existingQRCodeTransaction.transactionId];
+            }
+        }
+        
+        if (existingQRCodeTransaction && !(isExistingTransactionTemporary))
```

Comment:
This code does not cancel anymore if there is already an existing SAS transaction.

Reply:
why not:
```suggestion
        if ([existingTransaction isKindOfClass:MXQRCodeTransaction.class])
        {
             MXQRCodeTransaction *existingQRCodeTransaction = (MXQRCodeTransaction*)existingTransaction;

            if (existingQRCodeTransaction.state == MXQRCodeTransactionStateUnknown)
            {
                existingTransaction = nil;

                // Remove fake QR code transaction
                [self removeQRCodeTransactionWithTransactionId:existingQRCodeTransaction.transactionId];
            }
        }

        if (existingTransaction)
```